stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/MediaManager.kt

```diff
@@ -154,9 +161,13 @@ class SpeakerManager(
                     devices.filterIsInstance(defaultFallback::class.java)
                 }?.firstOrNull()
                 val fallback =
-                    defaultFallbackFromType ?: selectedBeforeSpeaker ?: devices.firstOrNull {
-                        it !is StreamAudioDevice.Speakerphone
-                    }
+                    defaultFallbackFromType
+                        ?: selectedBeforeSpeaker.takeIf {
+                            it !is StreamAudioDevice.Speakerphone && devices.contains(it)
+                        }
+                        ?: devices.firstOrNull {
+                            it !is StreamAudioDevice.Speakerphone
+                        }
```

Comment:
I feel the readability of the code is poor, its too complex to understand a simple code 

Can we improve it? Wdyt?

Maybe this 

```kotlin
val isSpeaker = selectedDevice.value is StreamAudioDevice.Speakerphone
val firstNonSpeaker = devices.firstOrNull { it !is StreamAudioDevice.Speakerphone }

selectedBeforeSpeaker = if (!isSpeaker) {
    selectedDevice.value
} else {
    firstNonSpeaker
}

val fallback: StreamAudioDevice? = when {
    defaultFallbackFromType != null -> defaultFallbackFromType
    selectedBeforeSpeaker != null &&
        selectedBeforeSpeaker !is StreamAudioDevice.Speakerphone &&
        devices.contains(selectedBeforeSpeaker) -> selectedBeforeSpeaker
    else -> firstNonSpeaker
}
```